### ISSUE TITLE
Possible error in storage cache

### ISSUE THREAD

**Issue by technosophos** [Azure/brigade#540]
I'm working on a feature to add `brig build list PROJECT_ID`, using `store.GetProjectBuilds()`. But I'm noticing that the first call to that API returns a different result than subsequent calls.

<img width="909" alt="screen shot 2018-07-09 at 16 35 36" src="https://user-images.githubusercontent.com/89193/42479497-5ee08280-8396-11e8-9a73-0598ec23b697.png">

I'm starting to hunt down the cause. But if anyone else has seen this, let me know. The relevant code is:

```go
		proj, err := store.GetProject(project_id)
		if err != nil {
			return err
		}

		fmt.Printf("getting builds for %s\n", proj.Name)

		bs, err = store.GetProjectBuilds(proj)
		if err != nil {
			return err
		}
```

**Comment by technosophos** [comment]
Okay, it looks like the issue is that when Brig uses the storage, it does not give the cache time to sync. So adding a BlockUntilAPICacheSynced works.

**Comment by technosophos** [comment]
Closed by #571
